### PR TITLE
[Spring-Cloud] Add a new parameter "--build-env" for app deploy

### PR DESCRIPTION
**Pull request by smile37773**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
spring-cloud

File: src/spring-cloud/HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 Release History
 ===============
+3.0.1
+---
+* New preview argument `az spring-cloud app deploy` has new argument "--build-env" to specify build module and jvm version and so on.
```
Comment (zhoxing-ms) [inline src/spring-cloud/HISTORY.md:5]:
```suggestion
* `az spring-cloud app deploy` has new preview argument "--build-env" to specify build module and jvm version and so on.
```

File: src/spring-cloud/azext_spring_cloud/_validators_enterprise.py [hunk 2]
```diff
@@ -36,6 +36,21 @@ def not_support_enterprise(cmd, namespace):
...
+
+
+def validate_target_module(cmd, namespace):
+    if namespace.resource_group and namespace.service and is_enterprise_tier(cmd, namespace.resource_group, namespace.service) and namespace.target_module is not None:
+        raise ClientRequestError("'--target-module' doesn't support for Enterprise tier Spring instance.")
+
+
+def validate_runtime_version(cmd, namespace):
+    if namespace.resource_group and namespace.service and is_enterprise_tier(cmd, namespace.resource_group, namespace.service) and namespace.runtime_version is not None:
+        raise ClientRequestError("'--runtime-version' doesn't support for Enterprise tier Spring instance.")
```
Comment (zhoxing-ms) [inline src/spring-cloud/azext_spring_cloud/_validators_enterprise.py:51]: Is error `ArgumentUsageError` more appropriate than error `ClientRequestError`?

Comment (smile37773) [reply src/spring-cloud/azext_spring_cloud/_validators_enterprise.py]: Sounds good. Updated.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**